### PR TITLE
replace Dsymbol.comment with hash table

### PR DESCRIPTION
**Pull request by WalterBright**
since according to https://github.com/dlang/dmd/pull/13762 it is 99.97% unused.

### CODE REVIEW

**Review by MoonlightSentinel — DISMISSED** [review]
The global state must be reset by  `dmd.frontend.deinitializeDMD`

**Comment by ibuclaw** [comment]
> since according to https://github.com/dlang/dmd/pull/13762 it is 99.97% unused.

How was this tested?

IIRC, comment is a dead field unless you are building ddoc.

File: src/dmd/dsymbol.d [hunk 1]
```diff
@@ -1347,7 +1382,7 @@ public:
     {
         //printf("ScopeDsymbol::syntaxCopy('%s')\n", toChars());
         ScopeDsymbol sds = s ? cast(ScopeDsymbol)s : new ScopeDsymbol(ident);
-        sds.comment = comment;
+        sds.addComment(comment);
```
Comment (ibuclaw) [inline src/dmd/dsymbol.d:1385]: I don't think this is needed as you have the property setter in place.

Comment (WalterBright) [reply src/dmd/dsymbol.d]: ok

**Review by RazvanN7 — APPROVED** [review]

**Comment by ibuclaw** [comment]
This PR introduced a regression.

https://issues.dlang.org/show_bug.cgi?id=24171

**Pull request merged**